src/cmd/promremotebench/query.go

```diff
@@ -63,25 +70,105 @@ func newQueryExecutor(opts queryExecutorOptions) *queryExecutor {
 	}
 }
 
-func (q *queryExecutor) Run() {
+func (q *queryExecutor) Run(checker Checker) {
 	q.Logger.Info("query load configured",
 		zap.Int("concurrency", q.Concurrency))
 	for i := 0; i < q.Concurrency; i++ {
-		go q.workerLoop()
+		go q.alertLoad(checker)
 	}
+
+	go q.accuracyCheck(checker)
 }
 
-func (q *queryExecutor) workerLoop() {
+// accuracyCheck checks the accuracy of data for one
+// host at a time.
+func (q *queryExecutor) accuracyCheck(checker Checker) {
+	type label struct {
+		name  string
+		value string
+	}
+	labels := make([]label, 0, len(q.Labels))
+	for k, v := range q.Labels {
+		labels = append(labels, label{name: k, value: v})
+	}
+
+	query := new(strings.Builder)
+	for i := 0; ; i++ {
+		func() {
+			// Exec in func to be able to use defer.
+			if i > 0 {
+				time.Sleep(q.Sleep)
+			}
+
+			query.Reset()
+			if q.Aggregation != "" {
+				mustWriteString(query, q.Aggregation)
+				mustWriteString(query, "({")
+			}
+
+			curHostnames := checker.GetHostNames()
+			if len(curHostnames) == 0 {
+				if i > 0 {
+					q.Logger.Error("no hosts returned in the checker, skipping accuracy check.")
+				}
+				return
+			}
+
+			var (
+				dps          []Datapoint
+				selectedHost string
+			)
+
+			for j := 0; j < 5; j++ {
+				selectedHost = curHostnames[rand.Intn(len(curHostnames))]
+				dps = checker.GetDatapoints(selectedHost)
+				if len(dps) > 1 {
```

Comment:
can you just make this an infinite loop and then you dont need to worry about checking `if len(dps) <= 1 && i > 1 {` below?

Reply:
Well I didn't want to do that because if we want to use this in an integration test one day, we don't want the test to be in an infinite loop. Also, you could turn on 100% new series, in which case this would just loop forever.